### ISSUE TITLE
TypedInput text does not transfer type or placeholder attributes

### ISSUE THREAD

**Issue by jeancarl** [node-red/node-red#2059]
### What are the steps to reproduce?

In a node configuration screen, I have an existing password input field with a placeholder attribute. 

```
<div class="form-row">
  <label for="node-config-input-taApiKey"><i class="fa fa-key"></i> API Key:</label>
  <input type="password" id="node-config-input-taApiKey" placeholder="Tone Analyzer API Key">
</div>   
```

<img width="460" alt="" src="https://user-images.githubusercontent.com/1245985/52885518-5cfb4480-3126-11e9-8661-2135469b07c4.png">

I have modified it to be a TypedInput field.

```
<div class="form-row">
  <label for="node-config-input-taApiKey"><i class="fa fa-key"></i> API Key:</label>
  <input type="password" id="node-config-input-taApiKey" placeholder="Tone Analyzer API Key">
  <input type="hidden" id="node-config-input-taUrl">
</div>   
```

### What happens?

The generated TypedInput field is a plain text input, erroneously displaying the value and not transferring the placeholder attribute value.

(without a value, text input is missing placeholder)
<img width="481" alt="screen shot 2019-02-15 at 1 37 42 pm" src="https://user-images.githubusercontent.com/1245985/52885701-f75b8800-3126-11e9-9b46-da91a1642b5e.png">

(value should be displayed as a password field, instead as plain text)
<img width="489" alt="screen shot 2019-02-15 at 1 37 36 pm" src="https://user-images.githubusercontent.com/1245985/52885709-fb87a580-3126-11e9-8ab8-cbfb22eaee3e.png">

### What do you expect to happen?

The placeholder and type attributes should be carried over to the TypedInput field:

(without a value, text input displays placeholder)
<img width="491" alt="screen shot 2019-02-15 at 1 36 40 pm" src="https://user-images.githubusercontent.com/1245985/52885765-2114af00-3127-11e9-87c7-9faa058d2c81.png">

(value is displayed as a password field)
<img width="488" alt="screen shot 2019-02-15 at 1 36 46 pm" src="https://user-images.githubusercontent.com/1245985/52885771-240f9f80-3127-11e9-8129-b852192c9110.png">

### Suggested fix

Please review suggested fix here:
https://github.com/jeancarl/node-red/commit/5e3a5c375648c15b5de29475d23a42cd85d958d9

### Please tell us about your environment:

- [ ] Node-RED version: 0.19.1
- [ ] node.js version: 8.11.2
- [ ] npm version: 6.4.1
- [ ] Platform/OS: Mac (latest)
- [ ] Browser: Firefox (latest)

**Comment by knolleary** [comment]
Hi @jeancarl 

this isn't a bug as such - using the TypedInput for a password field is simply not something it was written for. But I can see the utility in doing so - particularly if its a simple enough fix.

If you wanted to open a pull-request onto the `dev` branch we can test it out properly.

**Comment by jeancarl** [comment]
https://github.com/node-red/node-red/pull/2060 has been submitted

**Comment by limengsong** [comment]
Hi @jeancarl：
Hello 
Could you give an example of a placeholder on a TypedInput Widget?
Thank you ！

**Comment by knolleary** [comment]
@limengsong the example is in the issue description.

**Comment by limengsong** [comment]
@knolleary  
placeholder="Tone Analyzer API Key"  
this？

**Comment by knolleary** [comment]
Yes, that is the standard html attribute for adding a placeholder to an input - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#The_placeholder_attribute

**Comment by limengsong** [comment]
@knolleary 
Sorry, maybe my description is not accurate. 
Re-described as follows:
I am using a TypedInput Widget and want to add a placeholder to this.
![image](https://user-images.githubusercontent.com/26559533/59496501-219a3800-8ec4-11e9-82e0-df6acbdcda47.png)
In the TypedInput, the placeholder attribute was not found.
![image](https://user-images.githubusercontent.com/26559533/59496675-89508300-8ec4-11e9-8ff9-928d2e08f053.png)

**Comment by knolleary** [comment]
It really as simple as adding the `placeholder` attribute to the `<input>` element before you turn it into a typedInput.

**Comment by limengsong** [comment]
```   
<div class="form-row">
        <label for="node-input-payload"> <span data-i18n="text.value"></span></label>
        <input type="text" id="node-input-payload" placeholder="my placeholder" style="width:70%">
        <input type="hidden" id="node-input-payloadType">
</div>
  oneditprepare: function () {
            $("#node-input-payloadType").val(this.payloadType);
            $("#node-input-payload").typedInput({
                default: 'default',
                typeField: $("#node-input-payloadType"),
                types: [{
                    value: "default",
                    label: "Default",
                    icon: "icons/node-flow-contrib-mi-command/default.png"
                }, {
                    value: "datasource",
                    label: "DataSource",
                    icon: "icons/node-flow-contrib-mi-command/datasource.png",
                    hasValue: false

                }],
            });
            $("#node-input-payload").typedInput('type', this.payloadType);
        }
```
At runtime debugging, input type="text" id="node-input-payload" becomes input type="hidden" id="node-input-payload"

![image](https://user-images.githubusercontent.com/26559533/59499247-94f27880-8ec9-11e9-85ab-1e897d2fd17d.png)

**Comment by knolleary** [comment]
What version of Node-RED are you using?

**Comment by limengsong** [comment]
14 Jun 17:43:49 - [info] Node-RED version: v0.19.6
14 Jun 17:43:49 - [info] Node.js  version: v10.14.1
14 Jun 17:43:49 - [info] Windows_NT 10.0.17134 x64 LE

**Comment by knolleary** [comment]
You are asking about how to use placeholder on an issue that reports that the placeholder attribute is not handled property by the typedInput in version 0.19.1.

The issue is marked as fixed in the 0.20 release.

**Comment by limengsong** [comment]
@knolleary 
Thank you very much, I will try。
